src/formatting.rs

```diff
@@ -604,30 +604,41 @@ fn parse_crate(
 ) -> Result<ast::Crate, ErrorKind> {
     let input_is_stdin = input.is_text();
 
-    let mut parser = match input {
-        Input::File(file) => parse::new_parser_from_file(parse_session, &file),
-        Input::Text(text) => parse::new_parser_from_source_str(
+    let parser = match input {
+        Input::File(file) => Ok(parse::new_parser_from_file(parse_session, &file)),
```

Comment:
Actually, the same thing can happen to the file case: https://github.com/rust-lang/rust/blob/1839c144bc6db8c07945caacd449aa7f3a76cd47/src/libsyntax/parse/mod.rs#L211.

Should I use `maybe_source_file_to_parser` here as well?

Reply:
> Should I use maybe_source_file_to_parser here as well?

I think ideally we should, though I think it is a low priority compared to stdin case as panicking on the file case seems fine.

Reply:
Okay then, I'll leave the file case as-is (and it's still blocked on https://github.com/rust-lang/rust/pull/56658).